Documentation/devel/distribution.md

```diff
@@ -0,0 +1,147 @@
+
+**NOTE:** I'm not sure if distribution is the best name for this concept or it should be called with other names like "image reference". So I'd like to hear your thoughts.
+
+A distribution represents the method to fetch an image starting from an input string. It doesn't mandates the image types. Usually a distribution can only provide one image type but some can provide different image types. For example:
+
+* Some docker registries also provides OCI images.
+* Rkt can internally implement the docker distribution to natively fetch a docker/OCI image from a registry or an ACI converted by docker2aci.
+
+Before this concept was introduced, rkt used the ImageType concept, the ImageType was mapped to the input image string format and internally covered multiple concepts like distribution, transport and image type (hidden since all are now appc ACIs).
+
+The distribution concept is used as the primary information in the different layers of rkt (fetching but also for references in a CAS/ref store).
+
+## Distribution types
+
+The distribution can be logically split in two types:
+
+* Name based distribution
+* Location based distribution
+
+Name based distribution can be considered the first class distribution type as it'll be the primarly used distribution type.
+Location based distribution can be thought as a sub-distribution for Name based distribution or a fast alternative (but not covering the Name based distribution feature) to directly fetch an image from a location.
+
+A distribution is represented as an URI with the uri scheme as "cimd" and the opaque data and query parts as the distribution information (I started trying to use just URLs being an URI subset for location but distributions IMHO don't cleanly map to a resource locator but better to a name, [rfc3986](https://tools.ietf.org/html/rfc3986)) I'm calling them URIs instead of URNs because it's the suggested name from the rfc (and URNs are defined as having an urn scheme by rfc2141).
```

Comment:
I think we can settle on the "cimd" and expand here what it means. Also, I'm not sure what you mean with "the opaque data and query parts as the distribution information" (but I got the idea) so I think it may be better phrased.